### DISCUSSION TITLE
QTree: allow click event to be populated

### DISCUSSION THREAD

**Discussion by hermit99** [quasarframework/quasar#14311] [Ideas / Proposals] [answered]
I like the current QTree design that it's flexible to allow various components as content. In my use case, I would like my QTree nodes to be editable, eg. QInput as node. However, the current implementation is too strict/limited on click event handling. When a node is clicked, it stops and prevents the click event to be populated, hence we can't enter QInput properly. A [codepen example](https://codepen.io/hermit99/pen/ZEoELRE?editors=1010) shows the problem.

My suggestion is, in QTree line 635-638:

    function onExpandClick (node, meta, e, keyboard) {
       if (e !== void 0) {
         stopAndPrevent(e)
       }
Can we please remove line 636-638, i.e `stopAndPrevent(e)`?
If removing this would cause other side effects, can we please come up with some other ways to allow these kinds of flexibility?

My current work-around is using right click, but this really hurts user experience. So I'm hoping there could be better solutions.
Thanks!

**Comment by metalsadman** [comment, marked as answer]
pretty simple https://codepen.io/metalsadman/pen/wvjvEVK?editors=1010

**Comment by hermit99** [comment]
Thanks @metalsadman ! That's good to know.
